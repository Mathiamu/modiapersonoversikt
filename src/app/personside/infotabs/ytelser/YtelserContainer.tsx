import * as React from 'react';
import styled from 'styled-components';
import VisuallyHiddenAutoFokusHeader from '../../../../components/VisuallyHiddenAutoFokusHeader';
import theme from '../../../../styles/personOversiktTheme';
import { erModiabrukerdialog } from '../../../../utils/erNyPersonoversikt';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';
import useBrukersYtelser from './useBrukersYtelser';

const Styling = styled.section`
    > * {
        margin-bottom: 0.5rem;
    }
    ${theme.resetEkspanderbartPanelStyling};
`;

function YtelserContainer() {
    const { rettigheter, pending } = useBrukersYtelser();

    return (
        <Styling>
            {erModiabrukerdialog() && <VisuallyHiddenAutoFokusHeader tittel="Ytelser" />}
            {rettigheter}
            {!pending && rettigheter.length === 0 && <AlertStripeInfo>Ingen ytelser funnet for bruker</AlertStripeInfo>}
            {pending && <CenteredLazySpinner />}
        </Styling>
    );
}

export default YtelserContainer;
