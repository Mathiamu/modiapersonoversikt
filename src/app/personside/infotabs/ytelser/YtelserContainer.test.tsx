import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import * as React from 'react';
import YtelserContainer from './YtelserContainer';

beforeEach(() => {
    Date.prototype.getTime = jest.fn(() => 0);
    Date.parse = jest.fn(() => 0);
});

test('Om YtelserContainer matcher snapshot', () => {
    const resultat = renderer.create(
        <TestProvider>
            <YtelserContainer />
        </TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});
