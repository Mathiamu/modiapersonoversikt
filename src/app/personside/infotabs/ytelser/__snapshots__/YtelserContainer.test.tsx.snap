// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om YtelserContainer matcher snapshot 1`] = `
.c6 {
  overflow: auto;
  box-shadow: 0 0 0 0.07rem rgba(0,0,0,0.25);
  border-radius: .25rem;
  background-color: white;
}

.c6 [role='table'] {
  text-align: left;
  width: 100%;
}

.c6 [role='table'] thead,
.c6 [role='table'] .thead[role='rowgroup'] {
  border-bottom: 0.2rem solid rgba(0,0,0,0.15);
  font-weight: 600;
}

.c6 [role='table'] [role='cell'],
.c6 [role='table'] [role='columnheader'] {
  padding: 0.7rem;
  vertical-align: middle;
}

.c6 [role='table'] [role='columnheader'] {
  padding-bottom: 0.7rem;
}

.c6 [role='table'] [role='row']:nth-child(even) {
  background-color: rgba(0,0,0,0.1);
}

.c4 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
}

.c4 > h3 {
  margin-bottom: 1rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c5 dt {
  color: #645f5a;
}

.c5 > div {
  margin: 0.5625rem 0;
  padding-right: 0.7rem;
  width: 13rem;
  overflow-wrap: break-word;
}

.c8 span:not(:last-child):after {
  content: '.';
}

.c7 li:not(:first-child) {
  padding-top: 1rem;
  margin-top: 1rem;
  border-top: solid 0.0625rem #b7b1a9;
}

.c9 {
  margin-top: 1rem;
}

.c11 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
  white-space: nowrap;
}

.c11 > *:not(:last-child) {
  margin-bottom: 1rem;
}

.c10 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
  white-space: nowrap;
}

.c10 > *:not(:last-child) {
  margin-bottom: 1rem;
}

.c1 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.1),0 0 0 1px rgba(0,0,0,0.1);
  padding: .5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 > * {
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 {
  padding: .5rem;
}

.c0 > * {
  margin-bottom: 0.5rem;
}

.c0 .ekspanderbartPanel__innhold {
  padding: 0;
}

.c0 .ekspanderbartPanel__hode:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

<section
  className="c0"
>
  <article
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <article
          aria-describedby="Helt tilfeldig ID"
          className="c4"
        >
          <h3
            className="typo-ingress"
            id="Helt tilfeldig ID"
          >
            Sykepengetilfellet
          </h3>
          <dl
            className="c5"
          >
            <div>
              <dt
                className="typo-normal"
              >
                Sykemeldt fra og med
              </dt>
              <dd
                className="typo-element"
              >
                06.02.2019
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Forbrukte dager
              </dt>
              <dd
                className="typo-element"
              >
                23
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Maksdato
              </dt>
              <dd
                className="typo-element"
              >
                16.01.2019
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Untatt aktivitet
              </dt>
              <dd
                className="typo-element"
              >
                —
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Midlertidig stans
              </dt>
              <dd
                className="typo-element"
              >
                —
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Sanksjonperiode
              </dt>
              <dd
                className="typo-element"
              >
                19.01.2014 - 05.09.2014
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Ferieperioder
              </dt>
              <dd
                className="typo-element"
              >
                19.01.2014 - 14.01.2014
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Forsikring
              </dt>
              <dd
                className="typo-element"
              >
                837.00 NOK
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Gyldig forsikring
              </dt>
              <dd
                className="typo-element"
              >
                Nei
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Forsikringsperiode
              </dt>
              <dd
                className="typo-element"
              >
                —
              </dd>
            </div>
          </dl>
        </article>
        <article
          aria-describedby="Helt tilfeldig ID"
          className="c4"
        >
          <h3
            className="typo-ingress"
            id="Helt tilfeldig ID"
          >
            Sykmelding
          </h3>
          <dl
            className="c5"
          >
            <div>
              <dt
                className="typo-normal"
              >
                Periode
              </dt>
              <dd
                className="typo-element"
              >
                09.02.2014 - 11.05.2014
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Behandlingsdato
              </dt>
              <dd
                className="typo-element"
              >
                01.04.2018
              </dd>
            </div>
            <div>
              <dt
                className="typo-normal"
              >
                Sykmelder
              </dt>
              <dd
                className="typo-element"
              >
                Lommelegen AS
              </dd>
            </div>
          </dl>
          <div
            className="typo-normal c6"
          >
            <table
              role="table"
            >
              <thead
                role="rowgroup"
              >
                <tr
                  role="row"
                >
                  <th
                    role="columnheader"
                  >
                    Periode
                  </th>
                  <th
                    role="columnheader"
                  >
                    Gradering
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    28.07.2014 - 03.01.2014
                  </td>
                  <td
                    role="cell"
                  >
                    89%
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </article>
      </div>
      <article
        aria-describedby="Helt tilfeldig ID"
        className="c4"
      >
        <h3
          className="typo-ingress"
          id="Helt tilfeldig ID"
        >
          Arbeidssituasjon
        </h3>
        <dl
          className="c5"
        >
          <div>
            <dt
              className="typo-normal"
            >
              Arbeidsgiverperiode
            </dt>
            <dd
              className="typo-element"
            >
              Ja
            </dd>
          </div>
          <div>
            <dt
              className="typo-normal"
            >
              Arbeidskategori
            </dt>
            <dd
              className="typo-element"
            >
              Ærlig arbeid
            </dd>
          </div>
        </dl>
        <ol
          aria-label="Arbeidsgivere"
          className="c7"
        >
          <li>
            <dl
              className="c5"
            >
              <div>
                <dt
                  className="typo-normal"
                >
                  Arbeidsgiver
                </dt>
                <dd
                  className="typo-element"
                >
                  Bærlevåg Burger og Bitcoin
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Kontonummer
                </dt>
                <dd
                  className="typo-element"
                >
                  <span
                    className="c8"
                  >
                    <span>
                      8786
                    </span>
                    <span>
                      52
                    </span>
                    <span>
                      75970
                    </span>
                  </span>
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Inntekstsperiode
                </dt>
                <dd
                  className="typo-element"
                >
                  Månedssats
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Inntekt for perioden
                </dt>
                <dd
                  className="typo-element"
                >
                  39,691.00 NOK
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Refusjonstype
                </dt>
                <dd
                  className="typo-element"
                >
                  Ikke refusjon
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Refusjon til dato
                </dt>
                <dd
                  className="typo-element"
                >
                  12.03.2018
                </dd>
              </div>
              <div>
                <dt
                  className="typo-normal"
                >
                  Sykepenger fra og med
                </dt>
                <dd
                  className="typo-element"
                >
                  18.02.2018
                </dd>
              </div>
            </dl>
          </li>
        </ol>
        <button
          className="knapp c9 knapp--hoved"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Vis alle arbeidsforhold
        </button>
      </article>
    </div>
    <section
      className="c10"
    >
      <h4
        className="typo-ingress"
      >
        Utbetalinger på vent
      </h4>
      <div
        className="typo-normal c6"
      >
        <table
          role="table"
        >
          <thead
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <th
                role="columnheader"
              >
                Årsak
              </th>
              <th
                role="columnheader"
              >
                Utbetalingsperiode
              </th>
              <th
                role="columnheader"
              >
                Utbetalingsgrad
              </th>
            </tr>
          </thead>
          <tbody
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Pga mistenkelig oppførsel
              </td>
              <td
                role="cell"
              >
                13.10.2017 - 10.10.2018
              </td>
              <td
                role="cell"
              >
                62%
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Pga mistenkelig oppførsel
              </td>
              <td
                role="cell"
              >
                13.10.2017 - 10.10.2018
              </td>
              <td
                role="cell"
              >
                62%
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </section>
    <section
      className="c11"
    >
      <h4
        className="typo-ingress"
      >
        Utbetalinger
      </h4>
      <div
        className="typo-normal c6"
      >
        <table
          role="table"
        >
          <thead
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <th
                role="columnheader"
              >
                Reg.dato
              </th>
              <th
                role="columnheader"
              >
                Type
              </th>
              <th
                role="columnheader"
              >
                Periode
              </th>
              <th
                role="columnheader"
              >
                Utb.grad
              </th>
              <th
                role="columnheader"
              >
                Dagsats
              </th>
              <th
                role="columnheader"
              >
                Bruttobeløp
              </th>
              <th
                role="columnheader"
              >
                Arb.giver
              </th>
              <th
                role="columnheader"
              >
                Org.nummer
              </th>
            </tr>
          </thead>
          <tbody
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                25.06.2018
              </td>
              <td
                role="cell"
              >
                KONTOØVERFØRING
              </td>
              <td
                role="cell"
              >
                14.01.2018 - 15.07.2017
              </td>
              <td
                role="cell"
              >
                79%
              </td>
              <td
                role="cell"
              >
                705.00 NOK
              </td>
              <td
                role="cell"
              >
                466.00 NOK
              </td>
              <td
                role="cell"
              >
                Vedvik - Lunde
              </td>
              <td
                role="cell"
              >
                1234567890
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                25.06.2018
              </td>
              <td
                role="cell"
              >
                KONTOØVERFØRING
              </td>
              <td
                role="cell"
              >
                14.01.2018 - 15.07.2017
              </td>
              <td
                role="cell"
              >
                79%
              </td>
              <td
                role="cell"
              >
                705.00 NOK
              </td>
              <td
                role="cell"
              >
                466.00 NOK
              </td>
              <td
                role="cell"
              >
                Vedvik - Lunde
              </td>
              <td
                role="cell"
              >
                1234567890
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="alertstripe alertstripe--info"
      >
        <span
          aria-label="info"
          className="alertstripe__ikon"
        >
          <svg
            focusable="false"
            height="1.5em"
            kind="info-sirkel-fyll"
            viewBox="0 0 24 24"
            width="1.5em"
          >
            <g
              fill="none"
            >
              <path
                d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                fill="#669DB4"
              />
              <path
                d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
                fill="#FFF"
              />
            </g>
          </svg>
        </span>
        <div
          className="typo-normal alertstripe__tekst"
        >
          For detaljer om utførte utbetalinger, se Utbetalingsoversikten.
        </div>
      </div>
    </section>
  </article>
</section>
`;
